nemo/utils/exp_manager.py

```diff
@@ -309,8 +321,9 @@ def check_resume(
     elif len(end_checkpoints) > 0:
         if resume_past_end:
             if len(end_checkpoints) > 1:
-                raise ValueError(f"Multiple multiple checkpoints {end_checkpoints} that matches *end.ckpt.")
-            logging.info(f"Resuming from {end_checkpoints[0]}")
+                logging.warning(
+                    f"Multiple checkpoints {end_checkpoints} that matches *end.ckpt. This should only happen if using model parallelism."
+                )
```

Comment:
Is it possible to add a check to see if this is a model parallel checkpoint? Maybe even a dumb check if "mp_rank" in checkpoint_name OR if AppState has model parallel enabled.
I am weary of making a change in behaviour without a deprecation period.

Reply:
The issue is that exp_manager is called before DDP is initialized. So there's no awareness of model parallelism at this point.

Reply:
Your'e right though, we can check if mp_rank is there. I'll add that.

Reply:
Another alternative is to force model parallel users to set resume_past_end to True since you are effectively ignoring this flag when running in model parallel.